libraries/chain/backing_store/db_context_rocksdb.cpp

```diff
@@ -0,0 +1,745 @@
+   }
+
+   int32_t db_context_rocksdb::db_idx64_next(int32_t iterator, uint64_t& primary) {
+      return sec_lookup_i64.next_secondary(iterator, primary);
+   }
+
+   int32_t db_context_rocksdb::db_idx64_previous(int32_t iterator, uint64_t& primary) {
+      return sec_lookup_i64.previous_secondary(iterator, primary);
+   }
+
+   /**
+    * interface for uint128_t secondary
+    */
+   int32_t db_context_rocksdb::db_idx128_store(uint64_t scope, uint64_t table, account_name payer, uint64_t id,
+                           const uint128_t& secondary) {
+      return sec_lookup_i128.store(name{scope}, name{table}, payer, id, secondary);
+   }
+
+   void db_context_rocksdb::db_idx128_update(int32_t iterator, account_name payer, const uint128_t& secondary) {
+      sec_lookup_i128.update(iterator, payer, secondary);
+   }
+
+   void db_context_rocksdb::db_idx128_remove(int32_t iterator) {
+      sec_lookup_i128.remove(iterator);
+   }
+
+   int32_t db_context_rocksdb::db_idx128_find_secondary(uint64_t code, uint64_t scope, uint64_t table, const uint128_t& secondary,
+                                    uint64_t& primary) {
+      return sec_lookup_i128.find_secondary(name{code}, name{scope}, name{table}, secondary, primary);
+   }
+
+   int32_t db_context_rocksdb::db_idx128_find_primary(uint64_t code, uint64_t scope, uint64_t table, uint128_t& secondary,
+                                  uint64_t primary) {
+      return sec_lookup_i128.find_primary(name{code}, name{scope}, name{table}, secondary, primary);
+   }
+
+   int32_t db_context_rocksdb::db_idx128_lowerbound(uint64_t code, uint64_t scope, uint64_t table, uint128_t& secondary,
+                                uint64_t& primary) {
+      return sec_lookup_i128.lowerbound_secondary(name{code}, name{scope}, name{table}, secondary, primary);
+   }
+
+   int32_t db_context_rocksdb::db_idx128_upperbound(uint64_t code, uint64_t scope, uint64_t table, uint128_t& secondary,
+                                uint64_t& primary) {
+      return sec_lookup_i128.upperbound_secondary(name{code}, name{scope}, name{table}, secondary, primary);
+   }
+
+   int32_t db_context_rocksdb::db_idx128_end(uint64_t code, uint64_t scope, uint64_t table) {
+      return sec_lookup_i128.end_secondary(name{code}, name{scope}, name{table});
+   }
+
+   int32_t db_context_rocksdb::db_idx128_next(int32_t iterator, uint64_t& primary) {
+      return sec_lookup_i128.next_secondary(iterator, primary);
+   }
+
+   int32_t db_context_rocksdb::db_idx128_previous(int32_t iterator, uint64_t& primary) {
+      return sec_lookup_i128.previous_secondary(iterator, primary);
+   }
+
+   /**
+    * interface for 256-bit interger secondary
+    */
+   int32_t db_context_rocksdb::db_idx256_store(uint64_t scope, uint64_t table, account_name payer, uint64_t id,
+                           const uint128_t* data) {
+      const eosio::chain::key256_t& secondary = *reinterpret_cast<const eosio::chain::key256_t*>(data);
```

Comment:
strict aliasing violation.

Change to:
```c++
eosio::chain::key256_t secondary;
std::mempcy(secondary.data(), data, secondary.size());
```

The compiler is smart enough to vaporize the memcpy calls and not cause any actual copying, but it is conformant with the strict-aliasing rules.

Reply:
fixed